antarest/study/business/link_management.py

```diff
@@ -51,38 +56,63 @@ class LinkManager:
     def __init__(self, storage_service: StudyStorageService) -> None:
         self.storage_service = storage_service
 
-    def get_all_links(self, study: RawStudy, with_ui: bool = False) -> t.List[LinkInfoDTO]:
+    def get_all_links(self, study: RawStudy) -> t.List[LinkInfoDTOType]:
         file_study = self.storage_service.get_storage(study).get_raw(study)
-        result = []
+        result: t.List[LinkInfoDTOType] = []
+
         for area_id, area in file_study.config.areas.items():
-            links_config: t.Optional[t.Dict[str, t.Any]] = None
-            if with_ui:
-                links_config = file_study.tree.get(["input", "links", area_id, "properties"])
+            links_config = file_study.tree.get(["input", "links", area_id, "properties"])
+
             for link in area.links:
-                ui_info: t.Optional[LinkUIDTO] = None
-                if with_ui and links_config and link in links_config:
-                    ui_info = LinkUIDTO(
-                        color=f"{links_config[link].get('colorr', '163')},{links_config[link].get('colorg', '163')},{links_config[link].get('colorb', '163')}",
-                        width=links_config[link].get("link-width", 1),
-                        style=links_config[link].get("link-style", "plain"),
-                    )
-                result.append(LinkInfoDTO(area1=area_id, area2=link, ui=ui_info))
+                link_properties = links_config[link]
+
+                link_creation_data: t.Dict[str, t.Any] = {"area1": area_id, "area2": link}
+                link_creation_data.update(link_properties)
+
+                link_data: LinkInfoDTOType
+                if int(study.version) < 820:
```

Comment:
Due to this PR https://github.com/AntaresSimulatorTeam/AntaREST/pull/2156 we handle versions differently now. You should do this instead (this goes for every version check you did in this PR):
`StudyVersion.parse(study.version) < STUDY_VERSION_8_2`